dotCMS/src/test/java/com/dotmarketing/db/DataSourceStrategyProviderTest.java

```diff
@@ -0,0 +1,128 @@
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import com.zaxxer.hikari.HikariDataSource;
+import javax.sql.DataSource;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+
+/**
+ * Tests for {@link DataSourceStrategyProvider}
+ * @author nollymar
+ */
+@RunWith(DataProviderRunner.class)
+public class DataSourceStrategyProviderTest {
+
+    @DataProvider
+    public static Object[] testCases() {
+
+        return new String[]{
+                "DBProperties",
+                "SystemEnv",
+                "DockerSecret",
+                "Tomcat"
+        };
+    }
+
+    /**
+     * Method to test: {@link DataSourceStrategyProvider#get()}
+     * Test case: Verify this order is respected when DataSource credentials are obtained when a custom
+     * provider is not set:
+     *            2. db.properties file
+     *            3. System environment variables
+     *            4. Docker secrets
+     *            5. context.xml
+     *
+     * Expected result: A DataSource is returned using the right provider
+     * @param testCase
+     * @throws IllegalAccessException
+     * @throws InstantiationException
+     * @throws ClassNotFoundException
+     */
+    @UseDataProvider("testCases")
+    @Test
+    public void testGet(final String testCase)
+            throws IllegalAccessException, InstantiationException, ClassNotFoundException {
+
+        final DataSourceStrategyProvider provider = Mockito.spy(DataSourceStrategyProvider.class);
+        final SystemEnvironmentProperties properties = Mockito.mock(SystemEnvironmentProperties.class);
+        final DBPropertiesDataSourceStrategy dbStrategy = Mockito.mock(DBPropertiesDataSourceStrategy.class);
+        final SystemEnvDataSourceStrategy systemEnvStrategy = Mockito.mock(SystemEnvDataSourceStrategy.class);
+        final DockerSecretDataSourceStrategy dockerSecretStrategy = Mockito.mock(DockerSecretDataSourceStrategy.class);
+        final TomcatDataSourceStrategy tomcatDataSourceStrategy = Mockito.mock(TomcatDataSourceStrategy.class);
+
+        final HikariDataSource dummyDatasource = new HikariDataSource();
+
+        Mockito.when(provider.getSystemEnvironmentProperties()).thenReturn(properties);
+        Mockito.when(provider.getDBPropertiesInstance()).thenReturn(dbStrategy);
+        Mockito.when(provider.getSystemEnvDataSourceInstance()).thenReturn(systemEnvStrategy);
+        Mockito.when(provider.getDockerSecretDataSourceInstance()).thenReturn(dockerSecretStrategy);
+        Mockito.when(provider.getTomcatDataSourceInstance()).thenReturn(tomcatDataSourceStrategy);
+
+        if (testCase.equals("DBProperties")) {
+            Mockito.when(dbStrategy.existsDBPropertiesFile()).thenReturn(true);
+        } else {
+            Mockito.when(dbStrategy.existsDBPropertiesFile()).thenReturn(false);
+        }
+
+        if (testCase.equals("SystemEnv")){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5026767)